stan/math/rev/fun/log_sum_exp.hpp

```diff
@@ -70,7 +70,7 @@ inline auto log_sum_exp(const T& x) {
   return apply_vector_unary<T>::reduce(x, [](const auto& v) {
     auto arena_v = to_arena(v);
 
-    var res = log_sum_exp(value_of(arena_v));
+    var res = log_sum_exp(arena_v.val());
```

Comment:
`.val()` fails here. Run `./runTests.py test/unit/math/mix/fun/log_sum_exp_test`

It is the Eigen const bug or whatever. `val_op` works but I would just rather go with `value_of(arena_v)`.

Reply:
Oh weird, I think it was actually because of the line in the prim version of `log_sum_exp()`

```
const Eigen::Ref<const plain_type_t<decltype(v)>>& v_ref = v;
```

If we use `to_ref()` there instead then we can use `.val()`

```
const auto& v_ref = to_ref(v);
```

Reply:
Okay good enough for me.